controllers/homework.js

```diff
@@ -712,7 +765,7 @@ router.get('/:assignmentId', (req, res, next) => {
 				}),
 			);
 		}
-		Promise.all(promises).then(([submissions, course, courseGroups]) => {
+		Promise.all(promises).then(async ([submissions, course, courseGroups]) => {
```

Comment:
Please convert this in async await, without then.
Yes it is possible to put this async function in then block, but detect the flow how errors are handled is hard to read. We have also the problem that await is missed for Promise.all. It sporn the async process out of the try catch. If an error is throw we have a unhandled rejection.

(I know this is a long long method..)